File: ReportManager/Models/ServerConnectionModel.cs
using MongoDB.Bson;

namespace ReportManager.Models
{
    public class ServerConnectionModel
    {
        public ObjectId Id { get; set; }
        public string ServerName { get; set; }
        public int Port { get; set; }
        public string? Instance { get; set; }
        public string DbType { get; set; }
        public string? Username { get; set; }
        public string? Password { get; set; }
        public string AuthType { get; set; }
        public ObjectId OwnerID { get; set; }
        public OwnerType OwnerType { get; set; }

        // MongoDB
        public string? AuthSource { get; set; } = null;
        public string? ReplicaSet { get; set; } = null;
        public bool? UseTLS { get; set; } = null;
    }
}

File: ReportManager/ClientApp/src/models/serverconnectionmodel.js
class ServerConnectionModel {
    constructor() {
        this.id = ""; // ObjectId in MongoDB, but string representation in frontend
        this.serverName = "";
        this.port = 0;
        this.instance = ""; // Optional
        this.dbType = "";
        this.username = ""; // Optional
        this.password = ""; // Optional
        this.authType = "";
        this.ownerID = ""; // ObjectId in MongoDB, but string representation in frontend
        this.ownerType = ""; // Enum

        // MongoDB specific properties
        this.authSource = null; // Optional
        this.replicaSet = null; // Optional
        this.useTLS = null; // Optional boolean
    }
}